apps/src/storage/dataBrowser/dataVisualizer/Snapshot.jsx

```diff
@@ -0,0 +1,122 @@
+/* global ClipboardItem */
+import React from 'react';
+import PropTypes from 'prop-types';
+import {connect} from 'react-redux';
+import moment from 'moment';
+import msg from '@cdo/locale';
+import {html2canvas} from '@cdo/apps/util/htmlToCanvasWrapper';
+import BaseDialog from '@cdo/apps/templates/BaseDialog';
+import PendingButton from '@cdo/apps/templates/PendingButton';
+import * as dataStyles from '../dataStyles';
+
+class Snapshot extends React.Component {
+  static propTypes = {
+    chartTitle: PropTypes.string.isRequired,
+    selectedOptions: PropTypes.string.isRequired,
+    // Provided via Redux
+    tableName: PropTypes.string.isRequired,
+    projectName: PropTypes.string.isRequired
+  };
+
+  state = {
+    isSnapshotOpen: false,
+    isCopyPending: false,
+    isSavePending: false
+  };
+
+  handleOpen = () => this.setState({isSnapshotOpen: true});
+  handleClose = () => this.setState({isSnapshotOpen: false});
+
+  getPngBlob = () => {
+    const element = document.getElementById('snapshot');
+    const options = {
+      background: '#fff'
+    };
+    return html2canvas(element, options).then(canvas => {
+      return new Promise(function(resolve, reject) {
+        canvas.toBlob(function(blob) {
+          resolve(blob);
+        });
+      });
+    });
```

Comment:
I think async/await simplifies this down to:
```js
getPngBlob = async () => {
  const element = document.getElementByid('snapshot');
  const options = {background: '#fff'};
  const canvas = await html2canvas(element, options);
  const blob = await new Promise(resolve => canvas.toBlob(resolve));
  return blob;
};
```

Reply:
Yeah, wow using async/await here is so slick. thanks for the tip!+